### PR TITLE
Grade dialog

### PR DESCRIPTION
**Pull request by MMRoopa**
check the grade dialog

### CODE REVIEW

File: lib/gradeDialog/grade_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                  fontWeight: FontWeight.bold,
+                  fontFamily: 'Poppins',
+                ),
+                textAlign: TextAlign.center,
+                maxLines: 1,
+              ),
+            ),
+            Expanded(
+              child: Container(
+                margin: const EdgeInsets.fromLTRB(20.0, 20.0, 20.0, 2.0),
```
Comment (Shreyas-Mith) [inline lib/gradeDialog/grade_dialog.dart]: Whyis there a bottom padding of 2

File: lib/gradeDialog/grade_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+                      bool checked = (index == selectedIndex);
+                      bool lastIndex = (index == (_list.length - 1));
+                      return GestureDetector(
+                        onTap: () {
+                          setState(() {
+                            selectedIndex = index;
+                          });
+                        },
+                        child: Container(
+                          margin: const EdgeInsets.fromLTRB(7.0, 5.0, 7.0, 5.0),
```
Comment (Shreyas-Mith) [inline lib/gradeDialog/grade_dialog.dart]: Why is there a margin here, grid view spacing should be handled in gridDelegate

File: lib/gradeDialog/grade_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                                color: checked
+                                    ? const Color(0xFF2DAD9D)
+                                    : const Color(0xFFFFF8C9),
+                                shape: const RoundedRectangleBorder(
+                                  borderRadius:
+                                      BorderRadius.all(Radius.circular(15.0)),
+                                ),
+                                child: Padding(
+                                  padding: const EdgeInsets.fromLTRB(
+                                      10.0, 20.0, 10.0, 10.0),
```
Comment (Shreyas-Mith) [inline lib/gradeDialog/grade_dialog.dart]: Padding not required. Center the text to the container

File: lib/gradeDialog/grade_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+                                        ),
+                                      )
+                                    ]),
+                                  ),
+                                ),
+                              ),
+                              Positioned(
+                                bottom: 12,
+                                right: 12,
+                                child: Offstage(
```
Comment (Shreyas-Mith) [inline lib/gradeDialog/grade_dialog.dart]: bottom and right , set it to 8

File: lib/gradeDialog/grade_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+                                  child: RichText(
+                                    textAlign: TextAlign.center,
+                                    text: TextSpan(children: [
+                                      TextSpan(
+                                        text: _list[index],
+                                        style: const TextStyle(
+                                          fontFamily: 'poppins',
+                                          fontSize: 28.0,
+                                          fontWeight: FontWeight.bold,
+                                          color: Color(0XFF5F5200),
```
Comment (Shreyas-Mith) [inline lib/gradeDialog/grade_dialog.dart]: The grade text color should be white, when selected.

File: lib/gradeDialog/grade_dialog.dart [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+                                    ? const Color(0xFF2DAD9D)
+                                    : const Color(0xFFFFF8C9),
+                                shape: const RoundedRectangleBorder(
+                                  borderRadius:
+                                      BorderRadius.all(Radius.circular(15.0)),
+                                ),
+                                child: Padding(
+                                  padding: const EdgeInsets.fromLTRB(
+                                      10.0, 20.0, 10.0, 10.0),
+                                  child: RichText(
```
Comment (Shreyas-Mith) [inline lib/gradeDialog/grade_dialog.dart]: Use Autosize rich text

File: lib/gradeDialog/grade_dialog.dart [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+                                        style: const TextStyle(
+                                          fontFamily: 'poppins',
+                                          fontSize: 28.0,
+                                          fontWeight: FontWeight.bold,
+                                          color: Color(0XFF5F5200),
+                                        ),
+                                      ),
+                                      WidgetSpan(
+                                        child: Transform.translate(
+                                          offset: const Offset(3, -12),
```
Comment (Shreyas-Mith) [inline lib/gradeDialog/grade_dialog.dart]: change offset to Offset(2, -14)

File: lib/gradeDialog/grade_dialog.dart [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+                                          child: !lastIndex
+                                              ? const Text(
+                                                  'th',
+                                                  //superscript is usually smaller in size
+                                                  textScaleFactor: 1,
+                                                  style: TextStyle(
+                                                    fontSize: 14.0,
+                                                    color: Color(0XFF5F5200),
```
Comment (Shreyas-Mith) [inline lib/gradeDialog/grade_dialog.dart]: fontfamily and font weight missing

File: lib/gradeDialog/grade_dialog.dart [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+                                bottom: 12,
+                                right: 12,
+                                child: Offstage(
+                                  offstage: !checked,
+                                  child: Container(
+                                    decoration: const BoxDecoration(
+                                        color: Colors.white,
+                                        shape: BoxShape.circle),
+                                    child: const Icon(
+                                      Icons.check,
```
Comment (Shreyas-Mith) [inline lib/gradeDialog/grade_dialog.dart]: there is icon called check_circle, use that.
Remove container and change the icon color from Color(0xFF2DAD9D) to white.
Set the icon size to 20

**Review by Shreyas-Mith — CHANGES_REQUESTED** [review]

File: lib/gradeDialog/grade_dialog.dart (hunk 1 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: added EdgeInsets.only(left: 20.0, top: 20.0, right: 20.0),

File: lib/gradeDialog/grade_dialog.dart (hunk 2 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: updated mainAxisSpacing: 10,
                      crossAxisSpacing: 15, in gridDelegate
and deleted margin: const EdgeInsets.fromLTRB(7.0, 5.0, 7.0, 5.0),

File: lib/gradeDialog/grade_dialog.dart (hunk 3 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: removed the padding

File: lib/gradeDialog/grade_dialog.dart (hunk 4 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: done

File: lib/gradeDialog/grade_dialog.dart (hunk 5 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: changed the text color to white when selected

File: lib/gradeDialog/grade_dialog.dart (hunk 6 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: WidgetSpan for superscript is not supported for AutosizeText.rich

File: lib/gradeDialog/grade_dialog.dart (hunk 7 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: changed to Offset(2, -14)

File: lib/gradeDialog/grade_dialog.dart (hunk 8 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: added fontFamily: 'poppins',
                                                  fontWeight: FontWeight.w500,

File: lib/gradeDialog/grade_dialog.dart (hunk 9 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: added Icon(
                                    Icons.check_circle,
                                    color: Colors.white,
                                    size: 20.0,
                                  ),

File: lib/gradeDialog/grade_dialog.dart (hunk 3 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: wrapped with Center Widget

File: lib/gradeDialog/grade_dialog.dart (hunk 6 above)
Comment (MMRoopa) [reply lib/gradeDialog/grade_dialog.dart]: Added 2 AutoSizeText in a row with transform.translate

**Pull request merged**